src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/ArrayDataFlow.cs

```diff
@@ -769,6 +771,93 @@ public static void Test ()
 			}
 		}
 
+		class TestGetInterfacesDataflow
+		{
+			static void GetInterfacesOnInterfaces ([DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.Interfaces)] Type t)
+			{
+				t.GetInterfaces ()[0].RequiresInterfaces ();
+
+				Type[] interfaces = t.GetInterfaces ();
+				for (int i = 0; i < interfaces.Length; i++) {
+					interfaces[i].RequiresInterfaces ();
+				}
+
+				foreach (var i in t.GetInterfaces ())
+					i.RequiresInterfaces ();
+			}
+
+			static void GetAllOnAll ([DynamicallyAccessedMembers (DynamicallyAccessedMemberTypes.All)] Type t)
+			{
+				t.GetInterfaces ()[0].RequiresAll ();
+
+				Type[] interfaces = t.GetInterfaces ();
+				for (int i = 0; i < interfaces.Length; i++) {
+					interfaces[i].RequiresAll ();
+				}
+
+				foreach (var i in t.GetInterfaces ())
+					i.RequiresAll ();
```

Comment:
This test will fail with the Roslyn analyzer because within Roslyn this is modeled as IEnumerables? @agocke would you know if we can make Roslyn not to do that?